[ty] Correctly compare generic function subtyping/assignability regardless of TypeVar identity (#18634)
astral-sh/ruff: crates/ty_python_semantic/resources/mdtest/type_properties/is_subtype_of.md

reveal_type(A()(1))  # revealed: str
```

### Generic callables and TypeVar identity

> For generic callables, the identity of a TypeVar is not relevant for assignability checks, as long
> as the signatures are structurally compatible and the TypeVar bounds and constraints are equivalent.
> Two callables that differ only in the names of their TypeVars should be mutually assignable.

Reviewer: ```suggestion
For generic callables, the identity of a TypeVar is not relevant for subtyping, as long
as the signatures are structurally compatible and the TypeVar bounds and constraints are equivalent.
Two callables that differ only in the names of their TypeVars should be subtypes of each other.
```
